lib/nerve/reporter/zookeeper.rb

```diff
@@ -20,18 +21,20 @@ def initialize(service)
     end
 
     def start()
-      log.info "nerve: waiting to connect to zookeeper at #{@path}"
+      log.info "nerve: waiting to connect to zookeeper at #{@zk_hosts}"
       # Ensure that all Zookeeper reporters re-use a single zookeeper
-      # connection to any given connection string. Note that you will
-      # end up with a number of connections equal to the number of hosts in
-      # the connection string because the randomization in initialize
+      # connection to any given set of zk_hosts. Note that we round robin
+      # the connection pool so that we evenly spread the initial connections
       @@zk_pool_lock.synchronize {
+        @path = @zk_hosts.sort.rotate(rand(@zk_hosts.size)).join(',')
         unless @@zk_pool.has_key?(@path)
-          log.info "nerve: creating pooled connection at #{@path}"
+          log.info "nerve: creating pooled connection to #{@path}"
           @@zk_pool[@path] = ZK.new(@path)
```

Comment:
I suspect that the underlying zookeeper C client does its own address shuffling?  See e.g. https://github.com/apache/zookeeper/blob/trunk/src/c/src/zookeeper.c#L556